src/ui/public/index_patterns/_field_format/contentTypes.js

```diff
@@ -6,6 +6,10 @@ export default function contentTypesProvider(highlightFilter) {
   var types = {
     html: function (format, convert) {
       return function recurse(value, field, hit) {
+        if (typeof value === 'undefined') {
```

Comment:
Going a bit off topic and not massively important, but is "-" a good choice as this could be the legitimate contents of a string field? How about (empty) in grey/italics?